jobsdb/jobsdb.go

```diff
@@ -2205,36 +2215,59 @@ func (*HandleT) copyJobsDSInTx(txHandler transactionHandler, ds dataSetT, jobLis
 }
 
 func (*HandleT) doStoreJobsInTx(txHandler transactionHandler, ds dataSetT, jobList []*JobT) error {
-	var stmt *sql.Stmt
-	var err error
+	store := func() error {
+		var stmt *sql.Stmt
+		var err error
 
-	stmt, err = txHandler.Prepare(pq.CopyIn(ds.JobTable, "uuid", "user_id", "custom_val", "parameters", "event_payload", "event_count", "workspace_id"))
-	if err != nil {
-		return err
-	}
+		stmt, err = txHandler.Prepare(pq.CopyIn(ds.JobTable, "uuid", "user_id", "custom_val", "parameters", "event_payload", "event_count", "workspace_id"))
+		if err != nil {
+			return err
+		}
 
-	defer stmt.Close()
+		defer stmt.Close()
+		for _, job := range jobList {
+			eventCount := 1
+			if job.EventCount > 1 {
+				eventCount = job.EventCount
+			}
 
-	for _, job := range jobList {
-		eventCount := 1
-		if job.EventCount > 1 {
-			eventCount = job.EventCount
+			if _, err = stmt.Exec(job.UUID, job.UserID, job.CustomVal, string(job.Parameters), string(job.EventPayload), eventCount, job.WorkspaceId); err != nil {
+				return err
+			}
 		}
-
-		if _, err = stmt.Exec(job.UUID, job.UserID, job.CustomVal, string(job.Parameters), string(job.EventPayload), eventCount, job.WorkspaceId); err != nil {
-			return err
+		_, err = stmt.Exec()
+		return err
+	}
+	const (
+		savepointSql = "SAVEPOINT doStoreJobsInTx"
+		rollbackSql  = "ROLLBACK TO " + savepointSql
+	)
+	if _, err := txHandler.Exec(savepointSql); err != nil {
+		return err
+	}
+	err := store()
+	var e *pq.Error
+	if err != nil && errors.As(err, &e) {
```

Comment:
I would expect the following to work:
```go
if err != nil && errors.As(err, pq.Error)
```

hasn't that the case ?

Reply:
it is according to the [docs](https://pkg.go.dev/errors)

> As unwraps its first argument sequentially looking for an error that can be assigned to its second argument, which must be a pointer. If it succeeds, it performs the assignment and returns true. Otherwise, it returns false. The form

```
var perr *fs.PathError
if errors.As(err, &perr) {
	fmt.Println(perr.Path)
}
```